arangod/Agency/Constituent.cpp

```diff
@@ -98,57 +97,68 @@ term_t Constituent::term() const {
 }
 
 /// Update my term
-void Constituent::term(term_t t) {
+void Constituent::term(term_t t, std::string const& votedFor) {
   MUTEX_LOCKER(guard, _castLock);
-  termNoLock(t);
+  termNoLock(t, votedFor);
 }
 
-void Constituent::termNoLock(term_t t) {
+void Constituent::termNoLock(term_t t, std::string const& votedFor) {
   // Only call this when you have the _castLock
   _castLock.assertLockedByCurrentThread();
 
   term_t tmp = _term;
   _term = t;
-
-  if (tmp != t) {
-    LOG_TOPIC(INFO, Logger::AGENCY) << _id << ": changing term, current role:"
-      << roleStr[_role] << " new term " << t;
-
-    _cast = false;
-
-    if (!_votedFor.empty()) {
-      Builder body;
-      { VPackObjectBuilder b(&body);
-        std::ostringstream i_str;
-        i_str << std::setw(20) << std::setfill('0') << t;
-        body.add("_key", Value(i_str.str()));
-        body.add("term", Value(t));
-        body.add("voted_for", Value(_votedFor)); }
-      
-      TRI_ASSERT(_vocbase != nullptr);
-      auto ctx = transaction::StandaloneContext::Create(_vocbase);
-      SingleCollectionTransaction trx(ctx, "election", AccessMode::Type::WRITE);
-      
-      Result res = trx.begin();
-      if (!res.ok()) {
-        THROW_ARANGO_EXCEPTION(res);
-      }
-      
-      OperationOptions options;
-      options.waitForSync = _agent->config().waitForSync();
-      options.silent = true;
-      
-      OperationResult result;
+  std::string tmpVotedFor = _votedFor;
+  _votedFor = votedFor;
+
+  if (tmp != t || tmpVotedFor != votedFor) {
+    LOG_TOPIC(INFO, Logger::AGENCY) << _id << ": changing term or votedFor, "
+      << "current role:" << roleStr[_role] << " term " << t << " votedFor: "
+      << votedFor;
+
+    Builder body;
+    { VPackObjectBuilder b(&body);
+      std::ostringstream i_str;
+      i_str << std::setw(20) << std::setfill('0') << t;
+      body.add("_key", Value(i_str.str()));
+      body.add("term", Value(t));
+      body.add("voted_for", Value(_votedFor)); }
+
+    LOG_TOPIC(WARN, Logger::AGENCY) << body.toJson();
```

Comment:
Is this left over from debugging?  Seems strange / not self explanatory as to why it exists.

Reply:
Looks like it. Is removed.